desc/optimize/_constraint_wrappers.py

```diff
@@ -1018,7 +1035,11 @@ def jvp_unscaled(self, v, x, constants=None):
     @functools.partial(jit, static_argnames=("self", "op"))
     def _jvp_f(self, xf, dc, constants, op):
         Fx = getattr(self._constraint, "jac_" + op)(xf, constants)
-        Fx_reduced = Fx[:, self._unfixed_idx] @ self._Z
+        Fx_reduced = (
+            Fx[:, self._unfixed_idx]
+            @ self._D[self._unfixed_idx][:, self._unfixed_idx]
+            @ self._Z  # TODO: replace with self._unfixed_idx_mat
+        )
```

Comment:
I think this should be equivalent to
```
Fx @ self._unfixed_idx_mat[self._eq_idx]
```
but I'm not certain and I want another opinion